c/auth_libecc.c

```diff
@@ -0,0 +1,227 @@
+// clang-format off
+#include "common.h"
+
+#include "secp256r1.h"
+
+
+#include "ckb_consts.h"
+#if defined(CKB_USE_SIM)
+// exclude ckb_dlfcn.h
+#define CKB_C_STDLIB_CKB_DLFCN_H_
+#include "ckb_syscall_auth_sim.h"
+#else
+#include "ckb_syscalls.h"
+#endif
+
+#include "blake2b.h"
+#include "ckb_auth.h"
+#undef CKB_SUCCESS
+// clang-format on
+
+
+int validate_signature_secp256r1(void *prefilled_data, const uint8_t *sig,
+                              size_t sig_len, const uint8_t *msg,
+                              size_t msg_len, uint8_t *output,
+                              size_t *output_len) {
+    int err = 0;
+
+    if (*output_len < BLAKE160_SIZE) {
+        return ERROR_INVALID_ARG;
+    }
+    CHECK2(msg_len == BLAKE2B_BLOCK_SIZE, ERROR_INVALID_ARG);
+    CHECK2(sig_len == SECP256R1_DATA_SIZE, ERROR_INVALID_ARG);
+    const uint8_t *signature_ptr = sig;
+    const uint8_t *pub_key_ptr =  signature_ptr + SECP256R1_SIGNATURE_SIZE;
```

Comment:
In Schorr, we have defined the signature as: pubkey + sig. To maintain consistency, place pubkey at the beginning.